lib/_http_outgoing.js

```diff
@@ -929,7 +929,7 @@ function write_(msg, chunk, encoding, callback, fromEnd) {
   }
 
   let ret;
-  if (msg.chunkedEncoding && chunk.length !== 0) {
+  if (msg.chunkedEncoding && chunk.length !== 0 && msg.req.method !== 'HEAD') {
```

Comment:
We already check this with `_hasBody` above

Reply:
@ronag Is it a good step to add code that prevents the response body from being written in HEAD requests?